ASTNode.hpp

```diff
@@ -126,12 +126,29 @@ class ASTNode {
 
     return symbols.GetValue(var_id, token);
   }
-  void RunConditional([[maybe_unused]] SymbolTable &symbols) {
+  void RunConditional(SymbolTable &symbols) {
     // conditional statement is of the form "if (expression1) statment1 else
     // statement2" so a conditional node should have 2 or 3 children: an
     // expression, a statement, and possibly another statement run the first
     // one; if it gives a nonzero value, run the second; otherwise, run the
     // third, if it exists
+    assert(children.size() == 2 || children.size() == 3);
+
+    size_t curr_idx{0};
+    double condition = children[curr_idx].RunExpect(symbols);
+
+    curr_idx += 1;
+    if (condition != 0) {
+      children[curr_idx].Run(symbols);
+      return;
+    }
+
+    if (children.size() < 3){
+      return;
+    }
+
+    curr_idx += 1;
+    children[curr_idx].Run(symbols);
```

Comment:
I don't think this `curr_idx` variable is necessary, since we know with certainty which child contains which type of node. The compilers probably inlines this for us already, but replacing each usage of `curr_idx` with its value at that point would make it a bit clearer what's happening.